### ISSUE TITLE
Add Kafka Connect TrustStore

### ISSUE THREAD

**Issue by reidmeyer** [kafbat/kafka-ui#432]
### Issue submitter TODO list

- [X] I've searched for an already existing issues [here](https://github.com/kafbat/kafka-ui/issues)
- [X] I'm running a supported version of the application which is listed [here](https://github.com/kafbat/kafka-ui/blob/main/.github/SECURITY.md) and the feature is not present there

### Is your proposal related to a problem?

I am running a a kafka connect that I want to hook kafka-ui up to, but when I try to use what i've hooked up I get error
`500
PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`

### Describe the feature you're interested in

In provectus/kafka-ui, someone already added the functionality I'm looking for. https://github.com/provectus/kafka-ui/pull/2833/files

If there is agreement, a similar PR could be made here.

### Describe alternatives you've considered

I assume I could add environment variables somewhere to override the built in java truststore. If anyone has tried this, I'd be curious.

I see there is a kafka truststore.. does kafka-ui use this to communicate with kafka-connect as well; if so, I could add my ca there as well.

### Version you're running

42c236d

**Comment by fallen-up** [comment]
@reidmeyer i highly doubt you have tls-auth on kafka-connect. which means you'll be ok with the solution:
https://github.com/kafbat/kafka-ui/pull/422

**Comment by fallen-up** [comment]
@reidmeyer 
i have a case like this:
kafka-connect behind https, but with self-signed certificates.

1) options with truststore, keystore don't help
2) replacing global trustcacerts in /usr/lib/jvm/zulu17-ca/lib/security/ causes self-signed certificates in strimzi-kafka to break. and you need to add ca for each cluster.

fix https://github.com/kafbat/kafka-ui/pull/422 eliminates such problem.

**Comment by reidmeyer** [comment]
@fallen-up 

I have a nearly identical problem as you. I have a strimzi kafka-connect deployed behind https with a self signed cert. Typically I add the relevant CA to verify this cert to the truststore.

What action should I take to fix my situation after #422 is merged?

**Comment by fallen-up** [comment]
need to add
`KAFKA_CLUSTERS_0_SSL_VERIFYSSL: "false" #or another id of your cluster`

it's possible not to wait for merge and build locally for testing.

**Comment by reidmeyer** [comment]
so that turns off ssl verification for the kafka cluster.. and then the global is used for kafka connect? or does this turn off verifyssl for everything..?

isn't turning off ssl verification a bad thing..?

to me, the solution here seems like it would make more sense: https://github.com/provectus/kafka-ui/pull/2833
where you configure different truststores for each element you are connecting to.. indeed.. i'm not interested in the tls auth of this pr, but the option to configure a truststore for kafka-connect..

If you search in provectus/kafka-ui, you'll see they have CONNECT_SSL_TRUSTSTORE_LOCATION, SCHEMA_REGISTRY_SSL_TRUSTSTORE_LOCATION, KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_LOCATION. This makes sense to me. My Kafka and kafka-connect have different CAs to verify.

**Comment by Haarolean** [comment]
I wonder what makes you think we don't have this?

https://github.com/kafbat/kafka-ui/blob/main/documentation/compose/kafka-ssl-components.yaml#L30
https://github.com/kafbat/kafka-ui/commit/e26f9787d842ad56aefbc98c7ec6f9851fd35c1b

**Comment by reidmeyer** [comment]
Hi @Haarolean,

There's a bit of a misunderstanding, but I now see the solution, and indeed this issue should be closed.

I was looking for answers about configuring a truststore for kafka connect, not the keystore..

I only saw one truststore configuration "KAFKA_CLUSTERS_0_SSL_TRUSTSTORELOCATION" which i expected was only for connecting to Kafka itself, rather than all components, and thus I thought the kafka connect truststore configuration was missing. I thought this because I found the [PR](https://github.com/provectus/kafka-ui/pull/2833) i linked to above in which the KAFKA_CLUSTERS_0_KAFKACONNECT_0_TRUSTSTORELOCATION was added to the provectus project. Now, I see that in version 0.6.0 of the provectus project, the truststore working were changed to use one shared truststore, and kafbat obviously does the same.

<img width="903" alt="Screenshot 2024-06-09 at 11 09 18 AM" src="https://github.com/kafbat/kafka-ui/assets/16112576/51a8ef5f-1d00-412c-b1ec-a6a75b6f46ab">

I thus will add my additional CA to my truststore and that should work. Thank you for the responses.
